pkg/cli/debug/debug.go

```diff
@@ -798,7 +800,39 @@ func (o *DebugOptions) createPod(pod *corev1.Pod) (*corev1.Pod, error) {
 	if err := o.CoreClient.Pods(namespace).Delete(context.TODO(), name, *metav1.NewDeleteOptions(0)); err != nil && !kapierrors.IsNotFound(err) {
 		return nil, fmt.Errorf("unable to delete existing debug pod %q: %v", name, err)
 	}
-	return o.CoreClient.Pods(namespace).Create(context.TODO(), pod, metav1.CreateOptions{})
+	debugPod, err := o.CoreClient.Pods(namespace).Create(context.TODO(), pod, metav1.CreateOptions{})
+	if err != nil {
+		return nil, err
+	}
+	if err = o.checkStatusForFailure(debugPod); err != nil {
+		return nil, err
+	}
+	return debugPod, nil
+}
+
+func (o *DebugOptions) checkStatusForFailure(pod *corev1.Pod) error {
+	var err error
+	// check for early pod failures - such as node affinity errors that will be best caught here
+	// wait 3s for better failure message, otherwise return generic failure error
+	err = wait.PollImmediate(1*time.Second, 3*time.Second, func() (bool, error) {
+		pod, err = o.CoreClient.Pods(pod.Namespace).Get(context.TODO(), pod.Name, metav1.GetOptions{})
```

Comment:
unless i'm mistaken, in testing I saw as kube-apiserver was rolling out w/ defaultNodeSelector update, the podStatus=fail precedes the message/reason so was left with 
`error: debug pod failed status: ` instead of `error: debug pod failed status: Pod Predicate NodeAffinity failed`